internal/network/manager.go

```diff
@@ -145,27 +102,89 @@ func (m *Manager) startICE(isOffer bool, remoteUfrag, remotePwd string) error {
 	return nil
 }
 
-func (m *Manager) startSRTP() {
-	// Glue code until SRTP is a Conn.
-	go func() {
-		buf := make([]byte, receiveMTU)
-		for {
-			n, err := m.srtpEndpoint.Read(buf)
-			if err != nil {
-				return
-			}
-			m.handleSRTP(buf[:n])
+func (m *Manager) handleSRTP() {
+	for {
+		r, ssrc, err := m.srtpSession.AcceptStream()
+		if err != nil {
+			return
 		}
-	}()
+
+		go func() {
+			buf := make([]byte, receiveMTU)
+			for {
+				i, err := r.Read(buf)
+				if err != nil {
+					return
+				}
+
+				packet := &rtp.Packet{}
+				if err := packet.Unmarshal(append([]byte{}, buf[:i]...)); err != nil {
+					fmt.Println("Failed to unmarshal RTP packet")
+					return
+				}
+				fmt.Println("Unmarshalled:", ssrc)
+			}
+		}()
+	}
 }
 
-func (m *Manager) createContextSRTP(isOffer bool) error {
+func (m *Manager) handleSRTCP() {
+}
+
+func (m *Manager) startSRTP(isOffer bool) error {
 	keyingMaterial, err := m.dtlsConn.ExportKeyingMaterial([]byte("EXTRACTOR-dtls_srtp"), nil, (srtpMasterKeyLen*2)+(srtpMasterKeySaltLen*2))
 	if err != nil {
 		return err
 	}
 
-	return m.CreateContextSRTP(keyingMaterial, isOffer)
+	offset := 0
```

Comment:
Maybe we can consume the keying material inside the srtp package, or is it to WebRTC specific?

Reply:
Yea! I I don't see a reason to not move it inside.

I think for our new `*Session` stuff we can make it less flexible, we are just writing abstractions to make our lives easier.